src/content_mapblock.cpp

```diff
@@ -1513,7 +1513,50 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 
 			v3f pos = intToFloat(p, BS);
 
-			std::vector<aabb3f> boxes = n.getNodeBoxes(nodedef);
+			u8 neighbors = 0;
+
+			// locate possible neighboring nodes to connect to
+			if (f.node_box.type == NODEBOX_CONNECTED) {
```

Comment:
Nope. Not. I don't see how.

At issue is that content_mapblock.cpp is using a VoxelManip, and collision.cpp is using Map. That means two very different methods of accessing the 6 neighboring nodes, and so this turns either in to a terribly complex function, or just two functions again.

I've tried folding this into any of the logical objects but it just gets ugly, and the code itself right now is readable, logical and yeah, it works.

If anyone knows how to, please, PM me a patch ;)